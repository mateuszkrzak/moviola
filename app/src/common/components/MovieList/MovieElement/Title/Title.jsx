import React from 'react';
import PropTypes from 'prop-types';

import styles from './Title.scss';

class Title extends React.Component {
  static defaultProps = {
    title: 'No title provided',
  };

  render() {
    return <span styleName="title">{this.props.title} </span>;
  }
}

// const Title = props => <h3 styleName="title">{props.title} </h3>;
Title.defaultProps = {
  title: 'No title provided',
};
Title.propTypes = {
  title: PropTypes.string,
};

export default Title;
