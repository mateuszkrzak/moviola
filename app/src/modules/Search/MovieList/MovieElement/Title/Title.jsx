import React from 'react';
import CSSModules from 'react-css-modules';
import PropTypes from 'prop-types';

import styles from './Title.scss';

class Title extends React.Component {
  static defaultProps = {
    title: 'No title provided',
  };

  render() {
    return <span styleName="title">{this.props.title} </span>;
  }
}

// const Title = props => <h3 styleName="title">{props.title} </h3>;
Title.defaultProps = {
  title: 'No title provided',
};
Title.propTypes = {
  title: PropTypes.string,
};

export default CSSModules(Title, styles);
