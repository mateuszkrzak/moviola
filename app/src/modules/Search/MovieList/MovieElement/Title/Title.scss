.title {
  text-transform: uppercase;
  font-size: 16px;
  font-weight: 500;  
}
