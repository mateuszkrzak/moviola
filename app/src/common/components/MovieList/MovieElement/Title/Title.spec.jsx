import { shallow } from 'enzyme';
import React from 'react';
import Title from './Title';

test('Title component renders "No title provided" if no passed props', () => {
  const wrapper = shallow(<Title />);
  expect(wrapper).toMatchSnapshot();
});

test('Title component renders title', () => {
  const mockTitle = { title: 'Test title of movie' };
  const wrapper = shallow(<Title {...mockTitle} />);
  expect(wrapper).toMatchSnapshot();
});
