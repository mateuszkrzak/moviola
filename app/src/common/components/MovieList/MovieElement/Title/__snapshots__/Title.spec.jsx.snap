// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component renders "No title provided" if no passed props 1`] = `
<span
  styleName="title"
>
  No title provided
   
</span>
`;

exports[`Title component renders title 1`] = `
<span
  styleName="title"
>
  Test title of movie
   
</span>
`;
